src/unix/fs.c

```diff
@@ -1053,10 +1053,83 @@ static void uv__to_stat(struct stat* src, uv_stat_t* dst) {
 }
 
 
+static int uv__fs_statx(int fd,
+                        const char* path,
+                        int is_fstat,
+                        int is_lstat,
+                        uv_stat_t* buf) {
+  STATIC_ASSERT(UV_ENOSYS != -1);
+#ifdef __linux__
+  static int no_statx;
+  struct uv__statx statxbuf;
+  int dirfd;
+  int flags;
+  int mode;
+  int rc;
+
+  if (no_statx)
+    return UV_ENOSYS;
+
+  dirfd = -100; /* AT_FDCWD */
+  flags = 0; /* AT_STATX_SYNC_AS_STAT */
+  mode = 0xFFF; /* STATX_BASIC_STATS + STATX_BTIME */
+
+  if (is_fstat) {
+    dirfd = fd;
+    flags |= 0x1000; /* AT_EMPTY_PATH */
+  }
+
+  if (is_lstat)
+    flags |= 0x100; /* AT_SYMLINK_NOFOLLOW */
```

Comment:
What's the reason not to use the actual constants?

Reply:
They don't exist when you compile against older kernel headers and I didn't feel like defining `UV__*` namesakes when they're only used once. :-)

Reply:
Oh yeah, it makes sense. I think `AT_FDCWD` and `AT_SYMLINK_NOFOLLOW` should be fine (we're already using `AT_FDCWD`) but it's ok. Thanks!

Reply:
Fair point about `AT_FDCWD` and `AT_SYMLINK_NOFOLLOW`! I've updated the PR.